impl/src/main/java/org/ehcache/internal/persistence/DefaultLocalPersistenceService.java

```diff
@@ -213,6 +222,20 @@ static void destroy(String identifier, FileBasedPersistenceContext fileBasedPers
     }
   }
 
+  private static boolean recursiveDeleteDirectoryContent(File file) {
+    if (file.isDirectory()) {
+      File[] contents = file.listFiles();
+      if (contents.length > 0) {
+        for (File f : contents) {
+          if(!recursiveDelete(f)) {
+            return false;
```

Comment:
I wouldn't abort the deletion process if one file/folder could not be deleted. I'd store that failure in a boolean flag, go on with the deletion and return that flag at the end of the method.